### PR TITLE
Generic IDP sign in with Microsoft

### PR DESCRIPTION
**Pull request by samtstern**
If the Java looks OK I will add Kotlin.

### CODE REVIEW

File: auth/app/build.gradle [hunk 1]
```diff
@@ -46,15 +46,19 @@ dependencies {
     implementation 'com.android.support:design:28.0.0'
 
     // Firebase Authentication
-    implementation 'com.google.firebase:firebase-core:16.0.7'
-    implementation 'com.google.firebase:firebase-auth:16.1.0'
+    implementation 'com.google.firebase:firebase-core:16.4.0-SNAPSHOT'
```
Comment (samtstern) [inline auth/app/build.gradle]: This will obviously have to change before launch.

**Comment by samtstern** [comment]
@malcolmdeck FYI

File: auth/app/build.gradle [hunk 2]
```diff
@@ -46,15 +46,19 @@ dependencies {
...
+    implementation 'com.google.firebase:firebase-auth:16.4.0-SNAPSHOT'
 
     // Google Sign In SDK (only required for Google Sign In)
-    implementation 'com.google.android.gms:play-services-auth:16.0.1'
+    implementation 'com.google.android.gms:play-services-auth:16.4.0-SNAPSHOT'
+    api 'com.google.android.gms:play-services-auth-api-phone:16.4.0-SNAPSHOT'
```
Comment (samtstern) [inline auth/app/build.gradle]: Dunno what this is doing here?

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            }.addOnFailureListener {
+                e -> Log.w(TAG, "checkPending:onFailure", e)
+            }
+        } else {
+            Log.d(TAG, "pending: null")
+        }
+
+    }
+
+    fun signIn() {
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt]: Maybe this function could be private?

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+        val currentUser = auth.currentUser
+        updateUI(currentUser)
+
+        // Look for a pending auth result
+        val pending = auth.pendingAuthResult
+        if (pending != null) {
+            pending.addOnSuccessListener {
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt]: Can we have the argument in the same line as the opening brace?
```suggestion
            pending.addOnSuccessListener { authResult ->
```

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                authResult -> Log.d(TAG, "checkPending:onSuccess:$authResult")
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt]:
```suggestion
                Log.d(TAG, "checkPending:onSuccess:$authResult")
```

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            }.addOnFailureListener {
+                e -> Log.w(TAG, "checkPending:onFailure", e)
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt]: Same comment from above would apply here.

**Comment by samtstern** [comment]
@rosariopfernandes thanks!  I'll make sure to address your comments before release.

**Pull request merged**